### DISCUSSION TITLE
Add action buttons in Wizards

### DISCUSSION THREAD

**Discussion by SubhamKumar101201** [filamentphp/filament#9002] [Help]
Can i add some actions in the wizards like I have 4 steps in the 2nd step I want two buttons one which redirects to step 3 and another button which redirects to step 4 
So step 3 and 4 is related to step 2 with buttons, 
And how can i add extra action buttons or buttons inside a multi step wizards

**Comment by SubhamKumar101201** [comment]
?

**Reply by leandrocfe** [reply]
Custom buttons inside the step?

**Reply by SubhamKumar101201** [reply]
yes, how can a add a custom button inside a wizard ?
any idea or reference

**Reply by elbokh** [reply]
You can use the Actions form component inside the wizard form.

```
Step::make('step')
	->schema([
		\Filament\Forms\Components\Actions::make([
			\Filament\Forms\Components\Actions\Action::make('actionName')
				->action( # ACTION CODE # )
		]),
	]),
```

**Comment by KumarSubham101201** [comment]
Undefined type 'App\Filament\Resources\ContactResource\Widgets\Filament\Forms\Components\Actions\Action'.intelephense(1009)
Use of unknown class: 'App\Filament\Resources\ContactResource\Widgets\Filament\Forms\Components\Actions\Action'PHP(PHP0413)

**Reply by leandrocfe** [reply]
```php
Step::make('step')
    ->schema([
        \Filament\Forms\Components\Actions::make([
            \Filament\Forms\Components\Actions\Action::make('customAction')
                ->label('My custom action')
                ->action(function () {
                    dd('custom action');
                })
        ])
    ])
```

**Comment by GHOST117s** [comment]
in this iam stuck at validation 
as soon i click on the button it just simply moves to next step  without validation
can i prevent that ??????
i have multipe steps 

``` ->modalFooterActions([
Action::make('TakePayment')
    ->form([
       
            ]),
    ])

->modalFooterActions([
Action::make('Process Payment')
->action(function () {                          
    
 
})
->form([    
    Placeholder::make('Payment Status')
        ->content(function () {
            return "sucess";
        })
        ->inlineLabel(),
    

])->modalFooterActions([

]),
]),  Action::make('Cancel')
->cancelParentActions('Purchase Workshop modal')```
